src/lib/philips.ts

```diff
@@ -53,80 +54,113 @@ export const knownEffects = {
+    light: (args?: modernExtend.LightArgs & {hueEffect?: boolean; gradient?: true | {extraEffects: string[]}}) => {
+        args = {hueEffect: true, turnsOffAtBrightness1: true, ota: true, ...args};
+        if (args.hueEffect || args.gradient) args.effect = false;
+        if (args.color) args.color = {modes: ["xy", "hs"], ...(isObject(args.color) ? args.color : {})};
+        const result = modernExtend.light(args);
+        result.toZigbee.push(philipsTz.hue_power_on_behavior, philipsTz.hue_power_on_error);
+        if (args.hueEffect || args.gradient) {
+            result.toZigbee.push(philipsTz.effect);
+            const effects = ["blink", "breathe", "okay", "channel_change", "candle"];
+            if (args.color) effects.push("fireplace", "colorloop");
+            if (args.gradient) {
+                result.toZigbee.push(philipsTz.gradient_scene, philipsTz.gradient({reverse: true}));
+                result.fromZigbee.push(philipsFz.gradient);
+                effects.push("sunrise");
+                if (args.gradient !== true) {
+                    effects.push(...args.gradient.extraEffects);
                 }
-            });
+                result.exposes.push(
+                    // gradient_scene is deprecated, use gradient instead
+                    ...exposeEndpoints(e.enum("gradient_scene", ea.SET, Object.keys(gradientScenes)), args.endpointNames),
+                    ...exposeEndpoints(
+                        e
+                            .list("gradient", ea.ALL, e.text("hex", ea.ALL).withDescription("Color in RGB HEX format (eg #663399)"))
+                            .withLengthMin(1)
+                            .withLengthMax(9)
+                            .withDescription("List of RGB HEX colors"),
+                        args.endpointNames,
+                    ),
+                );
+                result.configure.push(async (device, coordinatorEndpoint, definition) => {
+                    for (const ep of device.endpoints.filter((ep) => ep.supportsInputCluster("manuSpecificPhilips2"))) {
+                        await ep.bind("manuSpecificPhilips2", coordinatorEndpoint);
+                    }
+                });
+            }
+            effects.push("finish_effect", "stop_effect", "stop_hue_effect");
+            result.exposes.push(...exposeEndpoints(e.enum("effect", ea.SET, effects), args.endpointNames));
         }
-        effects.push("finish_effect", "stop_effect", "stop_hue_effect");
-        result.exposes.push(...exposeEndpoints(e.enum("effect", ea.SET, effects), args.endpointNames));
-    }
-    return result;
-}
 
-export function philipsOnOff(args?: modernExtend.OnOffArgs) {
-    args = {powerOnBehavior: false, ota: true, ...args};
-    const result = modernExtend.onOff(args);
-    result.toZigbee.push(philipsTz.hue_power_on_behavior, philipsTz.hue_power_on_error);
-    return result;
-}
+        const customCluster = m.deviceAddCustomCluster("manuSpecificPhilips3", {
```

Comment:
@chrivers lot's of refactoring here, but this is basically the thing that is needed. Let me know if this is OK and I will merge it.

Reply:
Wow! That's amazing @Koenkk :D

Let me just give this a look!